DeDriftAndResample/MSMRemoveGroupDrift.sh

```diff
@@ -4,488 +4,326 @@
-	unset p_InputRegName  # InRegName
-	unset p_TargetRegName # TargetRegName
-	unset p_RegName                # RegName
-	unset p_HighResMesh            # HighResMesh
-	unset p_LowResMesh             # LowResMesh
-
-	# set default values
-
-	# parse arguments
-	local num_args=${#arguments[@]}
-	local argument
-	local index
-
-    for ((index = 0; index < num_args; ++index))
-    do
-		argument=${arguments[index]}
-
-		case ${argument} in
-			--help)
-				show_usage
-				exit 0
-				;;
-			--path=*)
-				p_StudyFolder=${argument#*=}
-				;;
-			--study-folder=*)
-				p_StudyFolder=${argument#*=}
-				;;
-			--subject-list=*)
-				p_Subjlist=${argument#*=}
-				;;
-			--common-folder=*)
-				p_CommonFolder=${argument#*=}
-				;;
-			--group-average-name=*)
-				p_GroupAverageName=${argument#*=}
-				;;
-			--input-registration-name=*)
-				p_InputRegName=${argument#*=}
-				;;
-			--target-registration-name=*)
-				p_TargetRegName=${argument#*=}
-				;;
-			--registration-name=*)
-				p_RegName=${argument#*=}
-				;;
-			--high-res-mesh=*)
-				p_HighResMesh=${argument#*=}
-				;;
-			--low-res-mesh=*)
-				p_LowResMesh=${argument#*=}
-				;;
-			*)
-				show_usage
-				log_Err_Abort "unrecognized option: ${argument}"
-				;;
-		esac
-	done
+#ARE THESE DESCRIPTIONS OK?????
+opts_AddMandatory '--study-folder' 'StudyFolder' 'path' "path to study folder"
 
-	local error_count=0
-	# check required parameters
-	if [ -z "${p_StudyFolder}" ]; then
-		echo "ERROR: path to study folder (--path= or --study-folder=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "p_StudyFolder: ${p_StudyFolder}"
-	fi
+opts_AddMandatory '--subject-list' 'Subjlist' 'list' "@ delimited list of subject ids"
 
-	if [ -z "${p_Subjlist}" ]; then
-		echo "ERROR: subject ID list (--subject-list=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "p_Subjlist: ${p_Subjlist}"
-	fi
+opts_AddMandatory '--common-folder' 'CommonFolder' 'path' "path to study folder"
 
-	if [ -z "${p_CommonFolder}" ]; then
-		echo "ERROR: common folder (--common-folder=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "p_CommonFolder: ${p_CommonFolder}"
-	fi
+opts_AddMandatory '--group-average-name' 'GroupAverageName' 'name' "output group average name (e.g. S900)"
 
-	if [ -z "${p_GroupAverageName}" ]; then
-		echo "ERROR: group average name (--group-average-name=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "p_GroupAverageName: ${p_GroupAverageName}"
-	fi
+opts_AddMandatory '--input-registration-name' 'InRegName' 'string' "Name of the input registration (e.g. MSMAll)"
 
-	if [ -z "${p_InputRegName}" ]; then
-		echo "ERROR: input registration name (--input-registration-name=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "p_InputRegName: ${p_InputRegName}"
-	fi
+opts_AddMandatory '--target-registration-name' 'TargetRegName' 'string' "Name of the target registration (e.g. MSMAll)"
```

Comment:
@glasserm This should probably use MSMSulc as the example.

Reply:
MSMSulc is the input reg name and MSMAll is the target reg name.  It looks like both now say MSMAll.